data/config-schema/src/main/resources/net/corda/schema/configuration/p2p.linkManager/1.0/corda.p2p.linkManager.json

```
@@ -91,10 +91,29 @@
     },
     "sessionsPerPeer": {
       "description": "The number of actively maintained sessions between two peers.",
-      "type": "integer",
-      "default": 4,
+      "type": ["integer", "null"],
+      "default": null,
       "minimum": 1
     },
+    "numOfSessionsPerPeer": {
+      "description": "The number of actively maintained sessions between two peers.",
+      "type": "object",
+      "default": {},
+      "properties": {
+        "forMgm": {
+          "description": "The number of actively maintained sessions between two peers if one of them is an MGM.",
+          "type": "integer",
+          "default": 1,
+          "minimum": 1
+        },
+        "forMembers": {
+          "description": "The number of actively maintained sessions between two peers if both peers are members.",
```

Might be nice to give a little more context. For example:

```suggestion
          "description": "The number of actively maintained sessions between two peers if both peers are members. The default value of 2 is recommended because the chance of both sessions failing on the same link manager is relatively low. If high-availability is critical, R3 recommended deploying multiple link manager instances.",
```

---

addressed in [980cb7f](https://github.com/corda/corda-api/pull/1307/commits/980cb7f1e2b40256508cee92a25f223b288ffb15)